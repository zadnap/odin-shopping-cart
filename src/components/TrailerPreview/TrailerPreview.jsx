import styles from './TrailerPreview.module.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlay, faStar } from '@fortawesome/free-solid-svg-icons';

function TrailerPreview({ title, year, rating, backdropSrc }) {
  return (
    <button
      className={styles.trailerPreview}
      style={{ backgroundImage: `url(${backdropSrc})` }}
    >
      <h3 className={styles.title}>{title}</h3>
      <div className={styles.info}>
        <p className={styles.year} aria-label="Year">
          {year}
        </p>
        <p className={styles.rating} aria-label="Rating">
          <FontAwesomeIcon icon={faStar} data-testid="star-icon" /> {rating}
        </p>
      </div>
      <div className={styles.playBtn}>
        <FontAwesomeIcon
          icon={faPlay}
          className={styles.icon}
          data-testid="play-icon"
        />
      </div>
    </button>
  );
}

export default TrailerPreview;
