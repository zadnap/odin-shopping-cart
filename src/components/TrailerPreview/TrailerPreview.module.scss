@use '@/styles/variables.scss' as *;

.trailerPreview {
  cursor: pointer;
  position: relative;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  padding: 10px;
  outline: none;
  border: none;
  border-radius: $border-radius;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 140px;
  width: 100%;
  color: $text-color;

  .title {
    position: absolute;
    font-size: 1.5rem;
    max-width: 90%;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .info {
    position: absolute;
    bottom: 10px;
    left: 10px;
    right: 10px;
    opacity: 0;
    transition: opacity ease 0.1s;
    font-size: 1.4rem;
    display: flex;
    justify-content: space-between;
    align-items: flex-end;

    .rating {
      color: yellow;
    }
  }

  .playBtn {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    opacity: 0;
    transition: opacity ease 0.1s;

    .icon {
      font-size: 3rem;
    }
  }

  &::before {
    position: absolute;
    content: '';
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    background-color: $overlay-color;
    border-radius: $border-radius;
  }

  &:hover {
    .title {
      text-decoration: underline;
    }
    .info,
    .playBtn {
      opacity: 1;
    }
  }
}
