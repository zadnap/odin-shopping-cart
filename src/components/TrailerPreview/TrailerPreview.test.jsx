import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import TrailerPreview from './TrailerPreview';

const mockProps = {
  id: 1,
  title: 'League of Legends: Arcane',
  duration: '2:45',
  rating: 9.0,
  backdropSrc:
    'https://image.tmdb.org/t/p/original/q8eejQcg1bAqImEV8jh8RtBD4uH.jpg',
};

describe('TrailerPreview component', () => {
  it('should render to match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <TrailerPreview {...mockProps} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
