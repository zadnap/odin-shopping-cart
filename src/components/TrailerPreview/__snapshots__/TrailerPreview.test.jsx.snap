// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TrailerPreview component > should render to match snapshot 1`] = `
<div>
  <a
    class="_trailerPreview_7373d7"
    data-discover="true"
    href="/trailers/1"
    style="background-image: url("https://image.tmdb.org/t/p/original/q8eejQcg1bAqImEV8jh8RtBD4uH.jpg");"
  >
    <h3
      class="_title_7373d7"
    >
      League of Legends: Arcane
    </h3>
    <div
      class="_info_7373d7"
    >
      <p
        aria-label="Duration"
        class="_duration_7373d7"
      >
        2:45
      </p>
      <p
        aria-label="Rating"
        class="_rating_7373d7"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star "
          data-icon="star"
          data-prefix="fas"
          data-testid="star-icon"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
         
        9
      </p>
    </div>
    <div
      class="_playBtn_7373d7"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-play _icon_7373d7"
        data-icon="play"
        data-prefix="fas"
        data-testid="play-icon"
        focusable="false"
        role="img"
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80L0 432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
          fill="currentColor"
        />
      </svg>
    </div>
  </a>
</div>
`;
